### ISSUE TITLE
Searching 73000-document-large index in Web causes Error: (Chrome 89)

### ISSUE THREAD

**Issue by daneroo** [jameslittle230/stork#158]
Your project is awesome!!!

I built a reasonably large index, ~73000 documents. (took about 48 minutes to build the index)

I filed this this issue as instructed in the console message:

If I find more details, I will amend or close and re-open a new issue.

This is error was in the developer tool console: (Chrome 89)
```
panicked at 'range end index 6 out of range for slice of length 5', src/index_versions/v3/search/entry_and_intermediate_excerpts.rs:151:33

panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: cannot recursively acquire mutex', /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys/wasm/../unsupported/mutex.rs:23:9
```

**Comment by jameslittle230** [comment]
Oh wow - you're hitting limits I've never seen before. Thanks for filing the issue, @daneroo.

If possible, could you send over a copy of your search index so I can debug? If that's not possible, maybe some statistics - how many files you're indexing, and what the total size of those files is?

Best,
James

**Comment by daneroo** [comment]
Wow yourself! That was a pretty fast response.
I mostly filed the issue because of the comment in the console.

I just started using stork this afternoon, so I'll give you details as I get them.

Also, the HTML widget is rendered in a React App (actually Next,js), so I am not sure if I got the details right. (Making sure the registration happens only once, etc... 
I'll have to look at your JS and css to understand it better.

== `tl;dr`
I am indexing a feed of my podcast listening history. 
The documents represent pages for each podcast and also one for each episode.

It worked for content of the last 45 days (1500 pages) - `.st`:4MB  - build: 12s. 

So I just pushed it to the max with 7 years of history - 73k pages - `.st`: 25MB - build: 48 minutes
I was surprised it worked at all with that size.
It makes for a rather large payload for the `.st` file too (`25MB`), so I'm not sure that make much sense from a UX perspective..

For now, I went back to the smaller index (last 45 days)
I am delighted at how well it works.

As soon as it's presentable, I can give you a public link to see it working.
Cheers,
Dan

**Comment by jameslittle230** [comment]
Gotcha - thanks for adding the details.

Not sure if this will be immediately helpful, but the Stork demo site is also a Next.js site, and [this component](https://github.com/stork-search/site/blob/master/src/components/stork.js) is what I wrote to make sure the index is only loaded once!

Can't wait to see it! Please let me know if you have any questions or feedback!

**Comment by jameslittle230** [comment]
I'm going to close this - I don't think it's totally actionable as it stands today. As Stork continues, I'll keep looking into improvements around supportable corpus sizes.
